src/infil.c

```diff
@@ -104,6 +104,18 @@ void infil_create(int subcatchCount, int model)
 //  Output:  none
 //
 {
+
+     // Allocate all 3 kinds of Infiltration data structures, because each
+     // subcatch[j] could be of any type. Doing this uses 3X the RAM: but the
+     // waste is only (2 X subcatchCount X 72 bytes) total.
+    HortInfil = (THorton *) calloc(subcatchCount, sizeof(THorton));
```

Comment:
@aerispaha, I don't like how you are allocating memory even when you may not use it. I think you can be more thoughtful with how you implement this in one allocation.

Reply:
I don't disagree that this is a bit inefficient - this current solution is a minimally invasive work around that wastes a bit of memory in the process. I'm thinking of two possible alternatives: 
1. tuck whatever infiltration object types are necessary (given the user-defined infiltration models) into an array of length equal to `subcatchCount` and retrieve these objects at each subcatchment index, something like `&InfiltrationObjects[j]`
2. continue to allocate memory for three separate infiltration structure arrays, each of length `subcatchCount`, but have `NULL` values at subcatchment indices where the infiltration type is not used, such that memory is saved. 

I think option 2 would require less changes to the source code (which may be a good thing), but option 1 may be more readable. I don't exactly know how to do either of these things in C, but I'm assuming they are possible and something I can figure out. 

What do you think?

Reply:
Even the largest SWMM5 models have less than 500,000 Subcatchments and most have less than 10,000.  Are you saving much memory given the small number of infiltration parameters?  Can't all infiltration data be allocated for all Subcatchments? this might allow mixed uses of infiltration types for LID's and Subcatchments or more than one pervious surface on a Subcatchment.

Reply:
@dickinsonre, if I understand things correctly, I think in your worst case 500,000 subcatchment scenario, we would be wasting at most 2 * 0.072kb * 500,000 = 72MB of memory. In the 10,000 subcatchment scenario we would be wasting at most 1.4MB. I'm not sure, but this does seem like a small amount of memory.